python/dlisio/ext/lis.cpp

```diff
@@ -320,15 +381,28 @@ noexcept (false) {
         dst = write_index( dst, fconf, index.index() );
         ++fmt;
     } else {
-        if ( fconf.mode == 1 ) {
+        if ( fconf.mode == 1 or fconf.samples > 1 ) {
             read_index( ptr, end, fconf, index );
         }
-        read_entry(fmt, ptr, end, dst, fconf.samples, fconf.framesize);
+        read_entry(fmt, ptr, end, dst, chstart, fconf.samples, fconf.framesize);
     }
 
+    // Skip back to the first subframe
+    dst = skip_frame( dst, fconf, -(fconf.samples - 1) );
+
```

Comment:
1. I am recording a complaint. I think all the dst jumps should be done in one outside method, not in the `read_entry`
2. Please, pleeeeease make the index reading code more clear. Do not mix fast channels and depth mode 1. I've been staring at these lines literally for an hour trying to figure out if they always work as they should. I still don't know.


```
read_frame:
  if depth_mode0 and no fast_channel:
     pass
  if depth_mode0 and fast_channel:
     read_index_from_disk_into_index //do not write into dst
  if depth_mode1 and no fast_channel:
     write_index
  if depth_mode1 and fast_channel:
     pass?
  if fast channel:
     for samples in samples // all, do not skip the last one, it's confusing
         write_subindex
 if depth_mode_1:
   update_the_spacing?

// we are at the start of channels that are of no interest for IndexChannel class
  while not_end_of_fmt:
     // do all the chst and jumps here, we have all the data
     for sample in samples // all other code that knows sampling rate is the same
       read_entry

read_entry:
   start_of_fast_channel_or_sample: do nothing, convenience character only
   end_of_fast_channel_or_sample: do nothing, convenience character only
```

Reply:
Refactored

Reply:
refactored